deployments/main.tf

```diff
@@ -37,15 +37,7 @@ module "dpe-sandbox-spacelift-development" {
   cluster_name = "dpe-k8-sandbox"
   vpc_name     = "dpe-sandbox"
 
-  vpc_cidr_block = "10.51.0.0/16"
-  # public_subnet_cidrs  = ["10.51.1.0/24", "10.51.2.0/24", "10.51.3.0/24"]
-  # private_subnet_cidrs = ["10.51.4.0/24", "10.51.5.0/24", "10.51.6.0/24"]
-  # azs                  = ["us-east-1a", "us-east-1b", "us-east-1c"]
-  # For now, we are only using one public and one private subnet. This is due to how
-  # EBS can only be mounted to a single AZ. We will need to revisit this if we want to
-  # allow usage of EFS ($$$$), or add some kind of EBS volume replication.
-  # Note: EKS requires at least two subnets in different AZs. However, we are only using
-  # a single subnet for node deployment.
+  vpc_cidr_block       = "10.51.0.0/16"
   public_subnet_cidrs  = ["10.51.1.0/24", "10.51.2.0/24"]
   private_subnet_cidrs = ["10.51.4.0/24", "10.51.5.0/24"]
   azs                  = ["us-east-1a", "us-east-1b"]
```

Comment:
I am leaving this on a 2 AZ deployment (Since it cannot be changed after the k8s cluster is created). I did set the single AZ part of `deployments/stacks/dpe-k8s-deployments/main.tf` to false so that the spot io cluster autoscaler can deploy EC2 instances to either private subnet

Reply:
No, This should work in conjunction with this change to limit the deployment to nodes in a specific zone, while still allowing nodes to spin up in either zone:

```
nodeSelector: {
  failure-domain.beta.kubernetes.io/zone: us-east-1a
}
```

Reply:
I also created https://sagebionetworks.jira.com/browse/IBCDPE-1097 as a follow-up here: Move worker node subnet off EKS cluster subnets

From these docs:  https://aws.github.io/aws-eks-best-practices/networking/subnets/#vpc-configurations


> Kubernetes worker nodes can run in the cluster subnets, but it is not recommended. During [cluster upgrades](https://aws.github.io/aws-eks-best-practices/upgrades/#verify-available-ip-addresses) Amazon EKS provisions additional ENIs in the cluster subnets. When your cluster scales out, worker nodes and pods may consume the available IPs in the cluster subnet. Hence in order to make sure there are enough available IPs you might want to consider using dedicated cluster subnets with /28 netmask.